sdk/cdn/Azure.ResourceManager.Cdn/src/Generated/Models/SecretProperties.cs

```diff
@@ -16,6 +16,17 @@ public SecretProperties()
         }
 
         /// <summary> object which contains secret parameters. </summary>
-        public SecretParameters Parameters { get; set; }
+        internal SecretParameters Parameters { get; set; }
+        /// <summary> The type of the Secret to create. </summary>
+        internal SecretType ParametersType
```

Comment:
I think `ParametersType` is accpetable as this property is internal but `SecretParametersType` may be better. Do we have any configuration that could modify the flattened single property name?

Reply:
we would do this with a rename directive

Reply:
- from: [specFile].json
    where: $.definitions.[modelName].properties.[propertyName]
    transform: $['x-ms-client-name'] = '[newPropertyName]'